lib/travis_event/listener.rb

```diff
@@ -0,0 +1,52 @@
+module TravisEvent
+  module Handlers
+    def self.handlers_for
+      Dir.glob("lib/travis_event/handlers/*.rb").collect do |f|
+        path = Pathname.new(f).relative_path_from(Pathname.new("lib")).to_s
+        path.chomp(".rb").classify.constantize
+      end
+    end
+  end
+
+  class Listener
+    ALL_EVENTS = %w(build:created build:started build:finished job:created job:started job:finished)
+
+    # repos: [Travis::Repository.find("rails/rails")]
+    def self.monitor(*repos)
+      Travis.listen(*repos) do |stream|
+        stream.on(*ALL_EVENTS) do |event|
+          args = extract_args(event)
+          if args
+            Handlers.handlers_for.each do |handler|
+              handler.perform_async(*args)
+            end
+          end
+        end
+      end
+    end
+
+    def self.extract_args(event)
+      # Event = Struct.new(:type, :repository, :build, :job, :payload)
+      number, state, build =
+        case event.type.split(':').first
+        when 'build'
+          [event.build.number, event.build.state, event.build]
+        when 'job'
+          [event.job.number, event.job.state, event.job.build]
+        else
+          Sidekiq::Logging.logger.info("#{name}##{__method__} Discarding unsupported event source type: #{event.type}")
+          return nil
+        end
+
+      branch_name = build.pull_request? ? build.pull_request_number : build.commit.branch
+      [event.repository.slug, number, event.type, state, branch_name, build.pull_request?]
```

Comment:
Instead of all this try:

```ruby
hash = event.to_h
hash.keys.each do |k|
  v = h.delete(k)
  v = v.attributes if v.respond_to?(:attributes)
  h[k.to_s] = v
end
```